src/base_decorator.rb

```diff
@@ -0,0 +1,11 @@
+require './nameable'
+class Decorator < Nameable
+  def initialize(_nameable)
+    super()
+    @nameable = Nameable
+  end
+
+  def correct_name
+    @nameable
+  end
```

Comment:
- [ ] it would be better to call `@nameable.correct_name` in order to call the correct name in Namable right now it will only return the `@nameable`